src/components/App.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import {Row, Col, Button, ButtonToolbar, ToggleButtonGroup, ToggleButton} from 'react-bootstrap';
+import {createStore} from 'redux';
+import reducer from '../reducers/index';
+import {Provider} from 'react-redux';
+import ExpressionsContainer from '../containers/ExpressionsContainer';
+import VariablesValueContainer from "../containers/VariablesValueContainer";
+import LanguageContainer from '../containers/LanguageContainer';
+import StructureContainer from '../containers/StructureContainer';
+import DownloadButton from './app/lib/DownloadButton';
+import {STUDENT_MODE, TEACHER_MODE} from "../constants";
+import {setMode} from "../actions";
+
+const store = createStore(reducer);
+
+// po kazdej zmene stavu sa vypise
+store.subscribe(() => {
+    let state = store.getState();
+    console.log('STATE:', state);
+});
+
+function importState(e) {
+    let file = e.target.files[0];
+    let fr = new FileReader();
+    fr.onload = function (e) {
+        store.dispatch({type: 'IMPORT_APP', content: e.target.result});
+    };
+    fr.readAsText(file);
+}
+
+function exportState() {
+    let state = store.getState();
+    let json = JSON.stringify({mode: state.mode, inputs: state.inputs, expressions: state.expressions});
+    return {
+        mime: 'application/json',
+        filename: 'export.json',
+        contents: json
+    }
+}
+
+const App = () => (
+    <Provider store={store}>
+        <div className={"app"}>
+            <div className='row import-export-bar'>
+                <div className='col-md-6'>
+                    <Button bsStyle='info' onClick={e => store.dispatch(setMode(STUDENT_MODE))}>Študent mód</Button>
```

Comment:
Rozmyslam, ako toto spravi trocha menej vyrazne a pouzitelne. Co by si povedal na takuto moznost:

[Input group](https://getbootstrap.com/docs/3.3/components/#input-groups):
 1. Label: Cvičenie:
 2. Input: Nazov cvicenia (zaroven nazov jsonoveho suboru, do ktoreho sa ulozi)
 3. Button: Uložiť (v slovenčine sa akcie na buttony píšu v neurčitku)
 4. Button: Importovať

Buttony by nemali az tak svietit, staci, ked budu oba secondary

Druhá input group – bez naozajstneho inputu, iba s labelom a dvoma gombikmi (ak sa to da; ak sa to neda, tak label a za nim [button group](https://getbootstrap.com/docs/3.3/components/#btn-groups)):
 1. Label: Mód:
 2. Button: Študent
 3. Button: Učiteľ

Gombik pre aktivny mod je active a primary.
Gombik pre neaktivny mod je secondary